tests/Microsoft.Identity.Test.Integration.net47/PoPCryptoProviderTests.cs

```diff
@@ -0,0 +1,179 @@
+            popConfig2.PopHttpMethod = HttpMethod.Post;
+
+            await confidentialApp.AcquireTokenForClient(s_keyvaultScope)
+                .WithProofOfPosession(popConfig2)
+                .ExecuteAsync(CancellationToken.None)
+                .ConfigureAwait(false);
+
+            Assert.AreEqual("PoP", popConfig2.PopAuthenticationRequestHeader.Scheme);
+            await VerifyPoPTokenAsync(
+                PublicCloudConfidentialClientID,
+                popConfig2).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// This calls a special endpoint that validates any POP token against a configurable HTTP request.
+        /// The HTTP request is configured through headers.
+        /// </summary>
+        private async Task VerifyPoPTokenAsync(string clientId, PopAuthenticationConfiguration popConfig)
+        {
+            var httpClient = new HttpClient();
+            HttpResponseMessage response;
+            var request = new HttpRequestMessage(HttpMethod.Post, PoPValidatorEndpoint);
+
+            request.Headers.Add("Secret", _popValidationEndpointSecret);
+            request.Headers.Add("Authority", "https://sts.windows.net/72f988bf-86f1-41af-91ab-2d7cd011db47/");
+            request.Headers.Add("ClientId", clientId);
+            request.Headers.Authorization = popConfig.PopAuthenticationRequestHeader;
+
+            // the URI the POP token is bound to
+            request.Headers.Add("ShrUri", popConfig.RequestUri.ToString());
+
+            // the method the POP token in bound to
+            request.Headers.Add("ShrMethod", popConfig.PopHttpMethod.ToString());
+
+            response = await httpClient.SendAsync(request).ConfigureAwait(false);
+
+            Assert.IsTrue(response.IsSuccessStatusCode);
+        }
+
+
+        private static Dictionary<string, string> GetTestSliceParams()
+        {
+            return new Dictionary<string, string>()
+            {
+                { "dc", "prod-wst-test1" },
+            };
+        }
+
+
+        private string _inMemoryCache = "{}";
+        private void ConfigureInMemoryCache(IPublicClientApplication pca)
+        {
+            pca.UserTokenCache.SetBeforeAccess(notificationArgs =>
+            {
+                byte[] bytes = Encoding.UTF8.GetBytes(_inMemoryCache);
+                notificationArgs.TokenCache.DeserializeMsalV3(bytes);
+            });
+
+            pca.UserTokenCache.SetAfterAccess(notificationArgs =>
+            {
+                if (notificationArgs.HasStateChanged)
+                {
+                    byte[] bytes = notificationArgs.TokenCache.SerializeMsalV3();
+                    _inMemoryCache = Encoding.UTF8.GetString(bytes);
+                }
+            });
+        }
+
+        private static X509Certificate2 GetCertificate()
+        {
+            X509Certificate2 cert = CertificateHelper.FindCertificateByThumbprint(TestConstants.AutomationTestThumbprint);
```

Comment:
This did not work on my machine :(

Reply:
I need to update the  default JWT algorithm to use "ES256" instead for the elliptic curve. 

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/8595a3eabcc2f2f2f53f9d6f8215f64686b9543d/src/client/Microsoft.Identity.Client/Internal/JsonWebTokenConstants.cs#L13